From b5e6763c4d05339b4ac2c953807634099cc6409b Mon Sep 17 00:00:00 2001
From: Benjamin Schimke <benjamin.schimke@canonical.com>
Date: Wed, 25 Jun 2025 09:40:02 +0200
Subject: [PATCH] Reduce go version requirement

---
 go.mod | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/go.mod b/go.mod
index 45bb4ee..22b004b 100644
--- a/go.mod
+++ b/go.mod
@@ -1,6 +1,6 @@
 module github.com/canonical/k8s-dqlite

-go 1.23.9
+go 1.23.8

 require (
 	github.com/canonical/go-dqlite/v2 v2.0.1
--
2.45.2
