From a044dd27e930d281875ea630e65843ef01193d91 Mon Sep 17 00:00:00 2001
From: Benjamin Schimke <benjamin.schimke@canonical.com>
Date: Sun, 20 Jul 2025 22:15:06 +0200
Subject: [PATCH] Make CGO configurable

---
 build | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/build b/build
index 894812858..3fa8cbeae 100755
--- a/build
+++ b/build
@@ -11,6 +11,8 @@ if [[ -n "${FAILPOINTS:-}" ]]; then
 	GIT_SHA="$GIT_SHA"-FAILPOINTS
 fi

+CGO_ENABLED="${CGO_ENABLED:-0}"
+
 # Set GO_LDFLAGS="-s" for building without symbols for debugging.
 GO_LDFLAGS="${GO_LDFLAGS:-} -X ${REPO_PATH}/version.GitSHA=${GIT_SHA}"

@@ -70,12 +72,12 @@ etcd_build() {

 	# Static compilation is useful when etcd is run in a container. $GO_BUILD_FLAGS is OK
 	# shellcheck disable=SC2086
-	CGO_ENABLED=0 go build ${GO_BUILD_FLAGS:-} \
+	CGO_ENABLED=${CGO_ENABLED} go build ${GO_BUILD_FLAGS:-} \
 		-installsuffix cgo \
 		-ldflags "$GO_LDFLAGS" \
 		-o "${out}/etcd" ${REPO_PATH} || return
 	# shellcheck disable=SC2086
-	CGO_ENABLED=0 go build ${GO_BUILD_FLAGS:-} \
+	CGO_ENABLED=${CGO_ENABLED} go build ${GO_BUILD_FLAGS:-} \
 		-installsuffix cgo \
 		-ldflags "$GO_LDFLAGS" \
 		-o "${out}/etcdctl" ${REPO_PATH}/etcdctl || return
@@ -96,7 +98,7 @@ tools_build() {
 	do
 		echo "Building" "'${tool}'"...
 		# shellcheck disable=SC2086
-		CGO_ENABLED=0 go build ${GO_BUILD_FLAGS} \
+		CGO_ENABLED=${CGO_ENABLED} go build ${GO_BUILD_FLAGS} \
 		  -installsuffix cgo \
 		  -ldflags "${GO_LDFLAGS}" \
 		  -o "${out}/${tool}" "${REPO_PATH}/${tool}" || return
--
2.45.2
